### ISSUE TITLE
Problem after updating Unity: missing prefabs

### ISSUE THREAD

**Issue by Fordesoft** [Seanba/SuperTiled2Unity#79]
Hi,

I upgraded from Unity 2018.2.16f1 to Unity 2019.3.0a3, and now my maps aren't showing up at all.

They show up as "Missing Prefab" in the scene hierarchy.  I'm also unable to drag the tmx files from the asset list into the scene hierarchy now.

I upgraded to the newest download of Super Tiled2Unity, but no effect.

I've tried right-clicking on the tmx files in the asset list and choosing Reimport.  That seems to have no effect at all.

How can I fix this?

**Comment by Seanba** [comment]
Hi there, @Fordesoft. Chances are decent that the Unity alpha build you are using has changed some dependencies on us. Are there are any errors in your console? (Press CTRL+SHIFT+C)

**Comment by Fordesoft** [comment]
A few, but it looks like they're all related to png files oddly enough.  Here's the error log:

![image](https://user-images.githubusercontent.com/46474996/58438699-bd8b0d80-8095-11e9-95d8-f4c499b33c51.png)

**Comment by Seanba** [comment]
I just installed 2019.3.0a3 and SuperTiled2Unity is working for me. Are you OSX? I'm on Windows so maybe there's a difference there.

When I run into Unity oddness related to upgrading versions I generally do the following and it fixes me up 90% of the time:

1. Shut down Unity
2. Delete my projects `Library` folder (its the folder that lives besides `Assets` and `ProjectSettings`
3. Re-open my project in Unity

This forces everything to be re-imported which may take some time.

**Comment by Fordesoft** [comment]
Nope, Windows 10 here too.  I did a Reimport All, and it looks like all of those errors in the log went away.  Still getting the Missing Prefab problem though.

Trying out your suggestion with the Library folder now.

**Comment by Fordesoft** [comment]
Still no luck there, unfortunately.  Prefab is still missing.

I'm going to try and see if a stable version of Unity 2019.1 will work.  Restoring a backup of my working 2018 project and upgrading to that now.

**Comment by Seanba** [comment]
If you have a backup of your 2018 project then you might have success with exporting it as a Unity Package and then installing that Unity Package to a new 2019 project.

**Comment by Fordesoft** [comment]
Alright, I think I've fixed it.  I got my project working in Unity 2019.1 at least -- haven't tried updating Super Tiled2Unity yet, but the old version of Super Tiled2Unity is working, so things are looking good so far.  I'm going to make a backup of the whole thing and then try upgrading Super Tiled2Unity.

Before I try that out...

I think I have some insight into what's causing the "missing prefab" issue in general now, insight that might be helpful for diagnosing the issue in the future:

It seems like the missing prefab issue happens whenever there's an error with *any* asset after an upgrade, not just errors with Super Tiled2Unity itself.  See, after I upgraded from 2018.2 to 2019.1 this time, I ran into the missing prefab issue.  Also had those PNG errors that I screenshotted above again (which went away after I did a Reimport All).  I also had a few errors with other assets, mainly TextMeshPro.

I googled around and resolved those TextMeshPro issues, and as soon as I did (and my error log was 100% clean), the Missing Prefab issue for Super Tiled2Unity went away.  Maybe Unity just doesn't want to do anything with the .tmx files until all those other compiler errors are resolved?  Weird.

I'm going to try upgrading Super Tiled2Unity to the newest version now.  If that goes well, which I think it will at this point, I'll close the issue.  Thanks for all your help so far, @Seanba .
